pkg/apis/clusteroperator/types.go

```diff
@@ -144,6 +144,19 @@ type AWSRegionAMIs struct {
 	MasterAMI *string
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterProviderConfigSpec is the cluster specification stored in the
+// ProviderConfig of a cluster.k8s.io Cluster.
+type ClusterProviderConfigSpec struct {
+	// +optional
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+
+	ClusterSpec
```

Comment:
Do we get any value from having two layers here, it feels like one might do, i.e. just embed our ClusterSpec (possibly renamed).

Reply:
At the moment, we are still using ClusterSpec in the cluster-operator Cluster. So I didn't want to introduce TypeMeta and ObjectMeta into that ClsuterSpec.